src/libs/actions/Report.js

```diff
@@ -1333,12 +1335,66 @@ function updateNotificationPreference(reportID, previousValue, newValue, navigat
             value: {notificationPreference: previousValue},
         },
     ];
-    API.write('UpdateReportNotificationPreference', {reportID, notificationPreference: newValue}, {optimisticData, failureData});
+    if (parentReportID && parentReportActionID) {
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`,
+            value: {[parentReportActionID]: {childReportNotificationPreference: newValue}},
+        });
+        failureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`,
+            value: {[parentReportActionID]: {childReportNotificationPreference: previousValue}},
+        });
+    }
+    API.write('UpdateReportNotificationPreference', {reportID, notificationPreference: newValue, parentReportID, parentReportActionID}, {optimisticData, failureData});
     if (navigate) {
         Navigation.goBack(ROUTES.REPORT_SETTINGS.getRoute(reportID));
     }
 }
 
+/**
+ * This will subscribe to an existing thread, or create a new one and then subsribe to it if necessary
+ *
+ * @param {String} childReportID The reportID we are trying to open
+ * @param {Object} parentReportAction the parent comment of a thread
+ * @param {String} parentReportID The reportID of the parent
+ * @param {String} prevNotificationPreference The previous notification preference for the child report
+ *
+ */
+function toggleSubscribeToChildReport(childReportID = '0', parentReportAction = {}, parentReportID = '0', prevNotificationPreference) {
+    if (childReportID !== '0') {
```

Comment:
this is weird to me. I think we should return early if childReportID is 0. I don't understand why anyone would want to subscribe to a thread prematurely i.e., when a thread does not exist yet.

Reply:
Yeah we talked about it briefly and I asked about it [here](https://expensify.slack.com/archives/C05RECHFBEW/p1697146420218079?thread_ts=1694794809.980809&cid=C05RECHFBEW) which led to this change. It's basically like slack where you can `Get notified about new replies`.